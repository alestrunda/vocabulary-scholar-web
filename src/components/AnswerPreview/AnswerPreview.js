import React from 'react';
import PropTypes from 'prop-types';
import NavigationCheck from 'material-ui/svg-icons/navigation/check';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import classNames from 'classnames';
import { isCorrectAnswer } from '../../misc';

const AnswerPreview = props => {
  const isCorrect = isCorrectAnswer(props.answered, props.word);

  return (
    <div
      className={classNames(
        'answer-preview',
        { 'answer-preview--correct': isCorrect },
        { 'answer-preview--wrong': !isCorrect },
        props.className
      )}
    >
      <div className="answer-preview__icon">
        {isCorrect && (
          <NavigationCheck color="#0f0" style={{ display: 'block' }} />
        )}
        {!isCorrect && (
          <NavigationClose color="#f00" style={{ display: 'block' }} />
        )}
      </div>
      <div className="answer-preview__answer">{props.word}</div>
      {props.answered && !isCorrect && (
        <div className="answer-preview__answered">
          You answered: <span className="text-italic">{props.answered}</span>
        </div>
      )}
    </div>
  );
};

AnswerPreview.propTypes = {
  className: PropTypes.string,
  word: PropTypes.string,
  answered: PropTypes.string,
};

export default AnswerPreview;
