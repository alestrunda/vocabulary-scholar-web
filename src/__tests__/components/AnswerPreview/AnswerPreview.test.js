import React from 'react';
import { shallow } from 'enzyme';

import AnswerPreview from '../../../components/AnswerPreview';

it('renders correctly if correct', () => {
  const props = {
    word: 'school',
    answered: 'school',
  };
  const root = shallow(<AnswerPreview {...props} />);
  expect(root).toMatchSnapshot();
});

it('renders correctly if wrong', () => {
  const props = {
    word: 'school',
    answered: 'car',
  };
  const root = shallow(<AnswerPreview {...props} />);
  expect(root).toMatchSnapshot();
});

it('renders correctly when no answer', () => {
  const props = {
    word: 'school',
    answered: '',
  };
  const root = shallow(<AnswerPreview {...props} />);
  expect(root).toMatchSnapshot();
});
