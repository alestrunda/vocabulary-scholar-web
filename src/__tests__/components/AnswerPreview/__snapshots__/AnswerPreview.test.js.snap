// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if correct 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnswerPreview
    answered="school"
    word="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="answer-preview__icon"
        >
          <NavigationCheck
            color="#0f0"
            style={
              Object {
                "display": "block",
              }
            }
          />
        </div>,
        <div
          className="answer-preview__answer"
        >
          school
        </div>,
        false,
      ],
      "className": "answer-preview answer-preview--correct",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavigationCheck
              color="#0f0"
              style={
                Object {
                  "display": "block",
                }
              }
            />,
            false,
          ],
          "className": "answer-preview__icon",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#0f0",
              "style": Object {
                "display": "block",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "school",
          "className": "answer-preview__answer",
        },
        "ref": null,
        "rendered": "school",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="answer-preview__icon"
          >
            <NavigationCheck
              color="#0f0"
              style={
                Object {
                  "display": "block",
                }
              }
            />
          </div>,
          <div
            className="answer-preview__answer"
          >
            school
          </div>,
          false,
        ],
        "className": "answer-preview answer-preview--correct",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavigationCheck
                color="#0f0"
                style={
                  Object {
                    "display": "block",
                  }
                }
              />,
              false,
            ],
            "className": "answer-preview__icon",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#0f0",
                "style": Object {
                  "display": "block",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "school",
            "className": "answer-preview__answer",
          },
          "ref": null,
          "rendered": "school",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly if wrong 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnswerPreview
    answered="car"
    word="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="answer-preview__icon"
        >
          <NavigationClose
            color="#f00"
            style={
              Object {
                "display": "block",
              }
            }
          />
        </div>,
        <div
          className="answer-preview__answer"
        >
          school
        </div>,
        <div
          className="answer-preview__answered"
        >
          You answered: 
          <span
            className="text-italic"
          >
            car
          </span>
        </div>,
      ],
      "className": "answer-preview answer-preview--wrong",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <NavigationClose
              color="#f00"
              style={
                Object {
                  "display": "block",
                }
              }
            />,
          ],
          "className": "answer-preview__icon",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#f00",
              "style": Object {
                "display": "block",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "school",
          "className": "answer-preview__answer",
        },
        "ref": null,
        "rendered": "school",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You answered: ",
            <span
              className="text-italic"
            >
              car
            </span>,
          ],
          "className": "answer-preview__answered",
        },
        "ref": null,
        "rendered": Array [
          "You answered: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "car",
              "className": "text-italic",
            },
            "ref": null,
            "rendered": "car",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="answer-preview__icon"
          >
            <NavigationClose
              color="#f00"
              style={
                Object {
                  "display": "block",
                }
              }
            />
          </div>,
          <div
            className="answer-preview__answer"
          >
            school
          </div>,
          <div
            className="answer-preview__answered"
          >
            You answered: 
            <span
              className="text-italic"
            >
              car
            </span>
          </div>,
        ],
        "className": "answer-preview answer-preview--wrong",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <NavigationClose
                color="#f00"
                style={
                  Object {
                    "display": "block",
                  }
                }
              />,
            ],
            "className": "answer-preview__icon",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#f00",
                "style": Object {
                  "display": "block",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "school",
            "className": "answer-preview__answer",
          },
          "ref": null,
          "rendered": "school",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You answered: ",
              <span
                className="text-italic"
              >
                car
              </span>,
            ],
            "className": "answer-preview__answered",
          },
          "ref": null,
          "rendered": Array [
            "You answered: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "car",
                "className": "text-italic",
              },
              "ref": null,
              "rendered": "car",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders correctly when no answer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnswerPreview
    answered=""
    word="school"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="answer-preview__icon"
        >
          
          <NavigationClose
            color="#f00"
            style={
              Object {
                "display": "block",
              }
            }
          />
        </div>,
        <div
          className="answer-preview__answer"
        >
          school
        </div>,
        "",
      ],
      "className": "answer-preview answer-preview--wrong",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <NavigationClose
              color="#f00"
              style={
                Object {
                  "display": "block",
                }
              }
            />,
          ],
          "className": "answer-preview__icon",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#f00",
              "style": Object {
                "display": "block",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "school",
          "className": "answer-preview__answer",
        },
        "ref": null,
        "rendered": "school",
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="answer-preview__icon"
          >
            
            <NavigationClose
              color="#f00"
              style={
                Object {
                  "display": "block",
                }
              }
            />
          </div>,
          <div
            className="answer-preview__answer"
          >
            school
          </div>,
          "",
        ],
        "className": "answer-preview answer-preview--wrong",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <NavigationClose
                color="#f00"
                style={
                  Object {
                    "display": "block",
                  }
                }
              />,
            ],
            "className": "answer-preview__icon",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#f00",
                "style": Object {
                  "display": "block",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "school",
            "className": "answer-preview__answer",
          },
          "ref": null,
          "rendered": "school",
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
